tests/org.eclipse.n4js.ide.tests/src/org/eclipse/n4js/ide/tests/server/AbstractIdeTest.java

```diff
@@ -0,0 +1,285 @@
+/*******************************************************************************
+ * Copyright (c) 2016 TypeFox GmbH (http://www.typefox.io) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *******************************************************************************/
+package org.eclipse.n4js.ide.tests.server;
+
+import java.io.File;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+import org.eclipse.lsp4j.ClientCapabilities;
+import org.eclipse.lsp4j.DidOpenTextDocumentParams;
+import org.eclipse.lsp4j.ExecuteCommandCapabilities;
+import org.eclipse.lsp4j.ExecuteCommandParams;
+import org.eclipse.lsp4j.InitializeParams;
+import org.eclipse.lsp4j.TextDocumentItem;
+import org.eclipse.lsp4j.WorkspaceClientCapabilities;
+import org.eclipse.n4js.cli.N4jscFactory;
+import org.eclipse.n4js.cli.N4jscTestFactory;
+import org.eclipse.n4js.cli.helper.N4jscTestLanguageClient;
+import org.eclipse.n4js.cli.helper.SystemOutRedirecter;
+import org.eclipse.n4js.ide.xtext.server.XLanguageServerImpl;
+import org.eclipse.n4js.projectDescription.ProjectType;
+import org.eclipse.n4js.projectModel.locations.FileURI;
+import org.eclipse.n4js.tests.codegen.Module;
+import org.eclipse.n4js.tests.codegen.Project;
+import org.eclipse.n4js.tests.codegen.Project.SourceFolder;
+import org.eclipse.n4js.utils.io.FileUtils;
+import org.eclipse.xtext.LanguageInfo;
+import org.eclipse.xtext.ide.server.UriExtensions;
+import org.eclipse.xtext.resource.IResourceServiceProvider;
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+
+import com.google.common.base.Strings;
+import com.google.inject.Inject;
+import com.google.inject.Injector;
+
+/**
+ * Signature help test class
+ */
+abstract public class AbstractIdeTest<T> {
+	static final String WORKSPACE_FOLDER = "/test-data";
+	static final String PROJECT_NAME = "test-project";
+	static final String MODULE_NAME = "MyModule";
+	static final String SRC_FOLDER = "src";
+	static final String FILE_EXTENSION = "n4js";
```

Comment:
Good idea but lets do that when we actually need it, since these values can also (and maybe better) be accessed from the `project` variable inside `performTest(File root, Project project, T t)`.